### PR TITLE
Worn icons some roles and derringer

### PR DESCRIPTION
**Pull request by KanohaShinobi**
adds .22 2 shot derringer and a 4 shot strip clip to load it, mayor spawns with it on their belt
sheriff spawns with a trail carbine and a 357
deputy just gets 357
bartender gets a sawn off on their belt slot (no ammo in pockets or anything just the gun, protect the bar!)
mayor gets his derringer with 2 clips for it (4 reloads)
worn icon defined in the update_icon of the guns (if it wasnt then got constant expression errors, they update on spawn anyway to my knowledge.)

I forgot the format for edits, but its good because it brings us closer to finalizing some jobs and the look of guns.

### CODE REVIEW

**Comment by KanohaShinobi** [comment]
LINTERS WHY I DIDNT EVEN COMMIT DIRECTORIES

File: mojave/items/guns/weapons/revolver.dm [hunk 1]
```diff
@@ -186,6 +203,14 @@
 	max_ammo = 5
 
 //Loaders
+/obj/item/ammo_box/ms13/derringer22
+	name = ".22 Derringer speed loader"
+	icon = 'mojave/icons/objects/ammo.dmi'
+	icon_state = "derringer"
+	ammo_type = /obj/item/ammo_casing/ms13/c22
+	caliber = "22"
```
Comment (ProfessorPopoff) [inline mojave/items/guns/weapons/revolver.dm:211]: caliber = "22" 

should be 

caliber = "..22"

File: mojave/items/guns/weapons/revolver.dm [hunk 2]
```diff
@@ -151,6 +163,11 @@
 	ammo_type = /obj/item/ammo_casing/shotgun/ms13/buckshot
 	max_ammo = 1
 
+/obj/item/ammo_box/magazine/internal/shot/ms13/derringer
+	name = "derringer internal tube"
+	ammo_type = /obj/item/ammo_casing/ms13/c22
```
Comment (ProfessorPopoff) [inline mojave/items/guns/weapons/revolver.dm:168]: Forgot to define the caliber. Insert 

caliber = ".22"

**Comment by ProfessorPopoff** [comment]
**Non-functioning worn icons**

/obj/item/gun/ballistic/rifle/ms13/hunting
/obj/item/gun/ballistic/rifle/ms13/hunting/scoped
/obj/item/gun/ballistic/automatic/ms13/marksman/battle
/obj/item/gun/ballistic/automatic/ms13/marksman/battle/rangemaster
/obj/item/gun/ballistic/shotgun/automatic/ms13/sks/scoped
/obj/item/gun/ballistic/shotgun/ms13/huntingshot

These are just the ones I found. I THINK i checked all the rifles/smgs, but that should be it.

**Comment by ProfessorPopoff** [comment]
You ought to also avoid roles spawning with guns in their hands if you can. Otherwise, you're probably going to have a lot of accidental firing early on.

**Comment by ProfessorPopoff** [comment]
/obj/item/storage/belt/holster/ gives the gunflip trait, and since you based your holster off that, it makes you flip guns whenever you interact with them, and doesn't really make much sense when you spin a shotgun by the trigger guard to reload it.

`/obj/item/storage/belt/holster/equipped(mob/user, slot)
	. = ..()
	if(slot == ITEM_SLOT_BELT)
		ADD_TRAIT(user, TRAIT_GUNFLIP, CLOTHING_TRAIT)`

File: mojave/items/guns/weapons/revolver.dm [hunk 3]
```diff
@@ -39,6 +39,18 @@
...
+/obj/item/gun/ballistic/revolver/ms13/derringer
+	name = "derringer"
+	desc = "A small and sneaky 2 shot pistol thats often concealed. This one's a bit fancy with some gold trim."
+	icon_state = "derringer"
+	inhand_icon_state = "derringer"
+	force = 5
+	mag_type = /obj/item/ammo_box/magazine/internal/shot/ms13/derringer
+	fire_sound = 'mojave/sound/ms13weapons/22pistol.ogg'
+	w_class = WEIGHT_CLASS_SMALL
+	extra_damage = 30
```
Comment (Hekzder) [inline mojave/items/guns/weapons/revolver.dm:52]:
```suggestion
	extra_damage = 25
```

For general consistency with the fact this is still just a .22 weapon, damage shouldn't be **double** the other two .22 weapons.

**Pull request merged**